DataFormats/BeamSpotAlpaka/interface/alpaka/BeamSpotAlpaka.h

```diff
@@ -0,0 +1,36 @@
+#ifndef DataFormats_BeamSpotAlpaka_interface_alpaka_BeamSpotAlpaka_h
+#define DataFormats_BeamSpotAlpaka_interface_alpaka_BeamSpotAlpaka_h
+
+#include <alpaka/alpaka.hpp>
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/memory.h"
+#include "DataFormats/BeamSpot/interface/BeamSpotPOD.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/memory.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+  class BeamSpotAlpaka {
+  public:
+    // default constructor, required by cms::alpakatools::Product<BeamSpotAlpaka>
+    BeamSpotAlpaka() = default;
+
+    // constructor that allocates cached device memory on the given CUDA stream
+    BeamSpotAlpaka(Queue& queue) : data_d_{cms::alpakatools::make_device_buffer<BeamSpotPOD>(queue)} {}
```

Comment:
I think this should use a `Queue const&`:
```suggestion
    BeamSpotAlpaka(Queue const& queue) : data_d_{cms::alpakatools::make_device_buffer<BeamSpotPOD>(queue)} {}
```